### ISSUE TITLE
[Bug] Spot instances don't work on GCP if the username has upper case letters

### ISSUE THREAD

**Issue by iRohith** [dstackai/dstack#975]
Steps to reproduce:

1. Register a user with dstack Sky that has an upper case letter, e.g. `iRohith`
2. Run `dstack run . -b gcp --spot`

It won't run even if GCP has spot capacity.

The error will log the following error:

```

ERROR 2024-03-10T11:29:03.545 dstack._internal.server.background.tasks.process_submitted_jobs (b625bd)fluffy-robin-1-0: got exception when launching g2-standard-8 in gcp/europe-west1
--
Traceback (most recent call last):
File "/usr/local/lib/python3.11/site-packages/dstack/_internal/server/background/tasks/process_submitted_jobs.py", line 227, in _run_job
launched_instance_info: LaunchedInstanceInfo = await run_async(
^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/dstack/_internal/server/utils/common.py", line 13, in run_async
return await asyncio.get_running_loop().run_in_executor(None, func_with_args)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
result = self.fn(*self.args, **self.kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dstack/dstack_cloud/services/backends/dstack/compute.py", line 69, in run_job
launched_instance_info = backend.compute().run_job(
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/dstack/_internal/core/backends/gcp/compute.py", line 167, in run_job
launched_instance_info = self.create_instance(instance_offer, instance_config)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/dstack/_internal/core/backends/gcp/compute.py", line 127, in create_instance
operation = self.instances_client.insert(request=request)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/google/cloud/compute_v1/services/instances/client.py", line 4164, in insert
response = rpc(
^^^^
File "/usr/local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
return wrapped_func(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 76, in error_remapped_callable
return callable_(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/google/cloud/compute_v1/services/instances/transports/rest.py", line 3197, in __call__
raise core_exceptions.from_http_response(response)
google.api_core.exceptions.BadRequest: 400 POST https://compute.googleapis.com/compute/v1/projects/dstack/zones/europe-west1-b/instances: Invalid value for field 'resource.labels': ''. Label value 'iRohith' violates format constraints. The value can only contain lowercase letters, numeric characters, underscores and dashes. The value can be at most 63 characters long. International characters are allowed.


```

**Comment by peterschmidt85** [comment]
When I'm trying the same on my machine (CLI 0.16.1, also dstack Sky, but with my user), the `auto` spot policy seems to work:

```
dstack run .
 Configuration          .dstack.yml
 Project                peterschmidt85
 User                   peterschmidt85
 Pool name              default-pool
 Min resources          2..xCPU, 8GB.., 1xGPU (24GB)
 Max price              -
 Max duration           1h
 Spot policy            auto
 Retry policy           no
 Creation policy        reuse-or-create
 Termination policy     destroy-after-idle
 Termination idle time  300s

 #  BACKEND  REGION       INSTANCE       RESOURCES                    SPOT  PRICE
 1  dstack   us-central1  g2-standard-4  4xCPU, 16GB, 1xL4 (24GB),    yes   $0.223804
                                         100GB (disk)
 2  dstack   us-east1     g2-standard-4  4xCPU, 16GB, 1xL4 (24GB),    yes   $0.223804
                                         100GB (disk)
 3  dstack   us-west1     g2-standard-4  4xCPU, 16GB, 1xL4 (24GB),    yes   $0.223804
                                         100GB (disk)
    ...
 Shown 3 of 191 offers, $5.876 max

Continue? [y/n]: y
witty-fly-1 provisioning completed (running)
```

Need to find out what I'm doing differently...

**Comment by peterschmidt85** [comment]
The most likely reason so fast seems to be that when using GCP's spot instances, there is an unhandled exception that prevents from any offers to be shown

**Comment by peterschmidt85** [comment]
Released in `0.16.3` ([Discord link](https://discord.com/channels/1106906313969123368/1163422692226113538/1217800685752422450))
